.snippets/code/tutorials/polkadot-sdk/parachains/zero-to-hero/add-pallets-to-runtime/pallets/custom-pallet/src/lib.rs

```diff
@@ -0,0 +1,190 @@
+#![cfg_attr(not(feature = "std"), no_std)]
+
+pub use pallet::*;
+
+#[cfg(test)]
+mod mock;
+
+#[cfg(test)]
+mod tests;
+
+#[frame_support::pallet(dev_mode)]
```

Comment:
is this being noted somewhere? `dev_mode` removes some constraints from FRAME macros, but then if you take the same code to a parachain template and try and apply it, you will get a bunch of compiler errors.

Reply:
I think the pallet actually doesn't need this -- you are already declaring weight and storage hashers, which are the main two things it relaxes.

Reply:
You are right, I'll just remove `dev_mode`